src/services/chat/index.ts

```diff
@@ -107,13 +113,26 @@ const plugin: FastifyPluginAsync<GraaspChatPluginOptions> = async (fastify) => {
           params: { itemId, messageId },
           body,
         } = request;
-        return await db.transaction(async (manager) => {
-          const member = notUndefined(user?.account);
-          const repositories = buildRepositories(manager);
-          const message = await chatService.patchOne(member, repositories, itemId, messageId, body);
-          await actionChatService.postPatchMessageAction(request, repositories, message);
-          return message;
-        });
+        try {
+          return await db.transaction(async (manager) => {
+            const member = notUndefined(user?.account);
+            const repositories = buildRepositories(manager);
+            const message = await chatService.patchOne(
+              member,
+              repositories,
+              itemId,
+              messageId,
+              body,
+            );
+            await actionChatService.postPatchMessageAction(request, repositories, message);
+            return message;
+          });
+        } catch (e: unknown) {
+          if (e instanceof EntryNotFoundAfterUpdateException || e instanceof EntityNotFound) {
+            throw new ChatMessageNotFound(messageId);
+          }
```

Comment:
Should this go a bit lower in the tree, like at the service level?

Reply:
Because it is an HTTP error, I'd catch and rethrow directly in the controller (because it's an HTTP controller). In this way, if we decided for some reason to add a graphQL endpoint for example, we won't mixed the HTTP errors with the graphQL errors (because they'll probably use the same services). But as always, I'm open to the discussion, so I dont know what do you think and if you prefer to throw in the service? @pyphilia, @spaenleh, @Defferrard

Reply:
To me, I think it's better to throw it in the controller!

Reply:
I agree with the justification for throwing in the controller.